curated_transformers/generation/stop.py

```diff
@@ -0,0 +1,100 @@
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING, List
+
+from torch import Tensor
+
+if TYPE_CHECKING:
+    from .state import GeneratorState
+
+
+class StopCondition(ABC):
+    """Base class for generation stop conditions."""
+
+    @abstractmethod
+    def update_completed(
+        self,
+        *,
+        state: "GeneratorState",
+        completed_exclude: Tensor,
+        completed_include: Tensor
+    ):
+        """
+        Update completed sequences according to the stop condition.
```

Comment:
Let's add a note to indicate the `Tensor` arguments are `out` parameters/modified in-place (not immediately clear currently).

Reply:
Updated docstring.